File: frontend/src/services/crypto.ts
import apiClient from './api';

export interface CryptoDepositIntent {
  deposit_id: string;
  network: string;
  currency: string;
  deposit_address: string;
  memo_or_tag?: string;
  amount_usdt: number;
  amount_kes: number;
  exchange_rate: number;
  qr_code_uri: string;
  expires_at_timestamp: number;
}

export interface CryptoVerifyResult {
  is_valid: boolean;
  tx_hash: string;
  amount_kes: number;
  new_balance: number;
  message: string;
}

export interface BinancePayOrder {
  prepay_id: string;
  checkout_url: string;
  qr_content: string;
  amount_usdt: number;
  amount_kes: number;
  currency: string;
  expire_time: number;
}

export const cryptoService = {
  createCryptoDeposit: async (data: { network: string; amount_kes?: number; amount_usdt?: number }): Promise<CryptoDepositIntent> => {
    const response = await apiClient.post<CryptoDepositIntent>('/payments/crypto/create-deposit', data);
    return response.data;
  },

  verifyCryptoTx: async (data: { deposit_id: string; tx_hash: string; network: string; amount_usdt: number }): Promise<CryptoVerifyResult> => {
    const response = await apiClient.post<CryptoVerifyResult>('/payments/crypto/verify', data);
    return response.data;
  },

  createBinanceOrder: async (data: { amount_kes: number }): Promise<BinancePayOrder> => {
    const response = await apiClient.post<BinancePayOrder>('/payments/binance/create-order', data);
    return response.data;
  },
};

File: backend/app/schemas/crypto.py
from decimal import Decimal
from typing import Optional
from pydantic import BaseModel, Field


class CryptoDepositRequest(BaseModel):
    network: str = Field(default="TRC20", description="TRC20, TON, or POLYGON")
    amount_kes: Optional[Decimal] = Field(None, gt=0, description="Amount in KES to deposit")
    amount_usdt: Optional[Decimal] = Field(None, gt=0, description="Amount in USDT to deposit")


class CryptoDepositResponse(BaseModel):
    deposit_id: str
    network: str
    currency: str
    deposit_address: str
    memo_or_tag: Optional[str] = None
    amount_usdt: Decimal
    amount_kes: Decimal
    exchange_rate: Decimal
    qr_code_uri: str
    expires_at_timestamp: int


class CryptoVerifyRequest(BaseModel):
    deposit_id: str
    tx_hash: str
    network: str = "TRC20"
    amount_usdt: Decimal


class CryptoVerifyResponse(BaseModel):
    is_valid: bool
    tx_hash: str
    amount_kes: Decimal
    new_balance: Decimal
    message: str


class BinancePayOrderRequest(BaseModel):
    amount_kes: Decimal = Field(..., gt=0, le=500000, description="Amount in KES to deposit via Binance Pay")


class BinancePayOrderResponse(BaseModel):
    prepay_id: str
    checkout_url: str
    qr_content: str
    amount_usdt: Decimal
    amount_kes: Decimal
    currency: str
    expire_time: int
